src/context/WalletProvider/NativeWallet/components/LegacyLogin.tsx

```diff
@@ -41,36 +36,67 @@ export const LegacyLogin = () => {
     handleSubmit,
     register,
     formState: { errors, isSubmitting },
+    reset,
   } = useForm({ shouldUnregister: true })
 
   const translate = useTranslate()
 
-  const isLoginError = (err: any): err is LoginError => err && typeof err.message === 'string'
+  const isLoginError = (err: any): err is LoginError =>
+    typeof err?.response?.data?.error?.msg === 'string' && typeof err.response.status === 'number'
 
-  const onSubmit = async (values: FieldValues) => {
-    try {
-      // @TODO: Replace this with API call when backend is ready
-      const { mnemonic } = await Promise.resolve({ mnemonic: DUMMY_MNEMONIC })
+  const isDecryptionError = (err: any): err is Error =>
+    typeof err?.message === 'string' && err.message.startsWith('Native wallet decryption failed')
 
-      if (!bip39.validateMnemonic(mnemonic)) {
-        // @TODO: Should we send an error to the user?
-      }
+  type LoginResponse = {
+    success: boolean
+    data: string
+    error: any
+  }
 
-      // TODO: use response to create wallet vault.
+  const MIGRATE_URL = getConfig().REACT_APP_WALLET_MIGRATION_URL
+
+  const onSubmit = async (values: FieldValues) => {
+    try {
+      const hashedPassword = await getPasswordHash(values.email, values.password)
+      const { data: response } = await axios.post<LoginResponse>(MIGRATE_URL, {
+        email: values.email,
+        password: hashedPassword,
+        twoFactorCode: values.twoFactorCode || undefined,
+      })
+      const { data: encryptedWallet } = response
       const vault = await Vault.create(undefined, false)
       vault.meta.set('createdAt', Date.now())
-      vault.set('#mnemonic', mnemonic)
+      vault.set(
+        '#mnemonic',
+        await decryptNativeWallet(values.email, values.password, encryptedWallet),
+      )
       history.push('/native/legacy/login/success', { vault })
+      // Clear the form state on success.
+      reset()
     } catch (err) {
-      if (isLoginError(err) && err.message === '2fa required') {
-        setTwoFactorRequired(true)
-
-        return
+      if (isLoginError(err)) {
+        if (err.response.status === 428 && err.response.data.error.msg === '2fa required') {
```

Comment:
I know this is something I did myself, but I would prefer adding these errors inside a `const` and comparing with the const instead of comparing with a string directly

Reply:
Could you elaborate? Do you think the messages belong in an enum? Or an object with a `key: { code, message }`?

Reply:
Could you elaborate? Do you think the messages belong in an enum? Or an object with a `key: { code, message }`?

Reply:
I was only thinking about the message, but linking the code with the message also looks fine to me, just avoiding using a raw string to compare

Reply:
Done.